### ISSUE TITLE
FFmpegFrameGrabber's start method does not run

### ISSUE THREAD

**Issue by hzhongl** [bytedeco/javacv#1295]
When I call the start method of FFmpegFrameGrabber, after running for a while, I find that the thread will block in the start method. If I don't continue running, I try to run it with the startUnsafe method. The same problem will occur after a while.
This is  my code.
` try(FFmpegFrameGrabber ff = FFmpegFrameGrabber.createDefault(filePath)){
            ff.setOption("stimeout" , "30000000");
            ff.startUnsafe();
            String rotate =ff.getVideoMetadata("rotate");
            Frame f;
            int i = 0;
            while (i <1) {
                f =ff.grabImage();
                IplImage src = null;
                if(null !=rotate &&rotate.length() > 1) {
                    OpenCVFrameConverter.ToIplImage converter =new OpenCVFrameConverter.ToIplImage();
                    src =converter.convert(f);
                    f =converter.convert(rotate(src, Integer.valueOf(rotate)));
                }
                doExecuteFrame(f,targetFilePath,targetFileName);
                i++;
            }
            ff.stop();`

**Comment by saudet** [comment]
If you're trying to play a file from the network, set the correct timeout option for your protocol:
https://ffmpeg.org/ffmpeg-protocols.html
https://github.com/bytedeco/javacv/blob/master/samples/FFmpegStreamingTimeout.java

**Comment by hzhongl** [comment]
> If you're trying to play a file from the network, set the correct timeout option for your protocol:
> https://ffmpeg.org/ffmpeg-protocols.html
> https://github.com/bytedeco/javacv/blob/master/samples/FFmpegStreamingTimeout.java
After I used the correct timeout setting, it didn't seem to work. Still blocking in the start method.

**Comment by saudet** [comment]
What does FFmpeg output to the console when you try this?

**Comment by hzhongl** [comment]
It will not block immediately, but after a period of normal operation, it will stop at the start method and will not continue to run. Even if other threads enter that method, they will be stuck in the start method and will not continue to run.

**Comment by saudet** [comment]
Ah, don't startUnsafe(), it's not safe. Use start().

**Comment by saudet** [comment]
Duplicate of #1139

**Comment by hzhongl** [comment]
I started with the start() method, not startUnsafe(), but I still have the above situation.

**Comment by saudet** [comment]
If one thread blocks in start(), it's normal for other threads to block in
start(). We need to unblock the first thread.

**Comment by hzhongl** [comment]
So what should I do? I have tried to set the timeout, thread interrupt and other methods can not be resolved.

**Comment by saudet** [comment]
Like I said, look at the (debug) log, it should tell us what is going on.

**Comment by hzhongl** [comment]
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Physical memory usage is too high: physicalBytes (976M) > maxPhysicalBytes (966M)
After I ran for an hour, it finally told me that this error occurred.

**Comment by hzhongl** [comment]
Caused by: java.lang.OutOfMemoryError: Physical memory usage is too high: physicalBytes (976M) > maxPhysicalBytes (966M)
        at org.bytedeco.javacpp.Pointer.deallocator(Pointer.java:585)
        at org.bytedeco.javacpp.Pointer.init(Pointer.java:125)
        at org.bytedeco.ffmpeg.avcodec.AVPacket.allocate(Native Method)
        at org.bytedeco.ffmpeg.avcodec.AVPacket.<init>(AVPacket.java:49)
        at org.bytedeco.javacv.FFmpegFrameGrabber.startUnsafe(FFmpegFrameGrabber.java:767)
        at org.bytedeco.javacv.FFmpegFrameGrabber.start(FFmpegFrameGrabber.java:758)

**Comment by saudet** [comment]
Increase that value if you expect to use more than that.

**Comment by hzhongl** [comment]
I found a solution. After setting these two parameters, I solved it.
System.setProperty("org.bytedeco.javacpp.maxphysicalbytes","0");
System.setProperty("org.bytedeco.javacpp.maxbytes","0");

**Comment by saudet** [comment]
BTW, I think I've fixed all the memory leaks occurring in `FFmpegFrameGrabber` and `FFmpegFrameRecorder`.
Please give it a try with the snapshots: http://bytedeco.org/builds/
